Demo/Common/IntegrationExampleComponents.swift

```diff
@@ -67,30 +106,53 @@ extension IntegrationExample {
                                                configuration: config)
         guard let presentableComponent = component else { return }
-        present(presentableComponent)
+        present(presentableComponent, delegate: self)
     }
 
+    // MARK: Convenience Store
+
     internal func presentConvenienceStore() {
-        guard let paymentMethod = paymentMethods?.paymentMethod(ofType: EContextPaymentMethod.self) else { return }
-        let component = EContextStoreComponent(paymentMethod: paymentMethod,
-                                               apiContext: ConfigurationConstants.apiContext,
-                                               configuration: BasicPersonalInfoFormComponent.Configuration(style: FormComponentStyle()))
-        present(component)
+        guard let component = convenienceStoreComponent(from: paymentMethods) else { return }
+        present(component, delegate: self)
+    }
+
+    internal func presentConvenienceStoreSession() {
+        guard let component = convenienceStoreComponent(from: sessionPaymentMethods) else { return }
+        present(component, delegate: session)
+    }
+
+    internal func convenienceStoreComponent(from paymentMethods: PaymentMethods?) -> EContextStoreComponent? {
+        guard let paymentMethods = paymentMethods,
+              let paymentMethod = paymentMethods.paymentMethod(ofType: EContextPaymentMethod.self) else { return nil }
+        return EContextStoreComponent(paymentMethod: paymentMethod,
+                                      apiContext: ConfigurationConstants.apiContext,
+                                      configuration: BasicPersonalInfoFormComponent.Configuration(style: FormComponentStyle()))
+    }
+
+    // MARK: BACS
+
+    internal func presentBACSDirectDebitComponent() {
+        guard let paymentMethod = paymentMethods?.paymentMethod(ofType: BACSDirectDebitPaymentMethod.self) else { return }
+        let component = BACSDirectDebitComponent(paymentMethod: paymentMethod,
+                                                 apiContext: ConfigurationConstants.apiContext)
+        bacsDirectDebitPresenter = BACSDirectDebitPresentationDelegate(bacsComponent: component)
+        component.presentationDelegate = bacsDirectDebitPresenter
+        present(component, delegate: self)
     }
 
     // MARK: - Presentation
 
-    private func present(_ component: PresentableComponent) {
+    private func present(_ component: PresentableComponent, delegate: (PaymentComponentDelegate & ActionComponentDelegate)?) {
         if let component = component as? PaymentAwareComponent {
             component.payment = payment
         }
 
         if let paymentComponent = component as? PaymentComponent {
-            paymentComponent.delegate = self
+            paymentComponent.delegate = delegate
         }
 
         if let actionComponent = component as? ActionComponent {
-            actionComponent.delegate = self
+            actionComponent.delegate = delegate
```

Comment:
actually Session has its own `actionComponent`, right? but this is harmless, and also useless in case of session :-)

Reply:
Exactly :)